core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -595,20 +596,22 @@ private List<ManifestFile> writeDataFileGroup(
   }
 
   protected List<ManifestFile> writeDeleteManifests(
-      Collection<DeleteFileHolder> files, PartitionSpec spec) {
+      Collection<DeleteFile> files, PartitionSpec spec) {
     return writeManifests(files, group -> writeDeleteFileGroup(group, spec));
   }
 
   private List<ManifestFile> writeDeleteFileGroup(
-      Collection<DeleteFileHolder> files, PartitionSpec spec) {
+      Collection<DeleteFile> files, PartitionSpec spec) {
     RollingManifestWriter<DeleteFile> writer = newRollingDeleteManifestWriter(spec);
 
     try (RollingManifestWriter<DeleteFile> closableWriter = writer) {
-      for (DeleteFileHolder file : files) {
+      for (DeleteFile file : files) {
+        Preconditions.checkState(
```

Comment:
I'm fine either way. My reasoning for using `checkState` initially was that something internally must have gotten wrong to reach that state state but I guess it's also possible that some other internal code just passes delete files that aren't a `PendingDeleteFile`.
Also getting this into a single line is hard, since the full line has 113 chars and I don't know what to omit from the error msg to make that fit into 100 chars